src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -47,32 +47,67 @@ public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, stri
                     return null;
                 }
                 // "ClCompile" for source files and "ClCompile" for header files
-                if (file.ItemType != "ClCompile")
+                // Don't rely on file extension to indentify header files since they can have any extension
+                var itemType = file.ItemType;
+                if (itemType != "ClCompile" && itemType != "ClInclude")
                 {
                     logger.WriteLine($"File's \"Item type\" is not supported. File: '{absoluteFilePath}'");
                     return null;
                 }
                 var platformName = ((VCPlatform)vcConfig.Platform).Name; // "Win32" or "x64"
                 var vcFileConfig = file.GetFileConfigurationForProjectConfiguration(vcConfig);
-                // We don't support custom build tools. VCCLCompilerTool is needed for all the necessary compilation options to be present
-                if (!(vcFileConfig.Tool is VCCLCompilerTool))
+                IVCRulePropertyStorage vcFileSettings;
+                if (itemType == "ClCompile")
```

Comment:
I think it's worth extracting this to a `GetVcFileSettings` method that returns IVCRulePropertyStorage. The method is getting quite big, and the comment regarding IVCRulePropertyStorage can be placed a doc comment.